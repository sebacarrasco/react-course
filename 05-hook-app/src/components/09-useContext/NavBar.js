import React from 'react';
import { NavLink } from 'react-router-dom';

// La diferencia entre Link y NavLink es que navlink acepta el atributo
// activeClassName, que permita agregar una clase al link cuando
// su ruta estÃ¡ activa.

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">

                <NavLink className="navbar-brand" to="/">Usecontext App</NavLink>

                <div className="collapse navbar-collapse" id="navbarNav">
                    
                    <div className="navbar-nav">
                        <NavLink className="nav-item nav-link" activeClassName="active" exact to="/">Home</NavLink>
                    
                        <NavLink className="nav-item nav-link" activeClassName="active" exact to="./about">About</NavLink>
                    
                        <NavLink className="nav-item nav-link" activeClassName="active" exact to="./login">Login</NavLink>
                    </div>
                </div>

            </div>
        </nav>
    )
}
