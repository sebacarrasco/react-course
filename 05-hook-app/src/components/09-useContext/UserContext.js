import { createContext } from "react";

// Sirve para compartir información entre varios componentes que
// no necesariamente son hijos, sino que pueden ser hermanos también
export const UserContext = createContext(null);