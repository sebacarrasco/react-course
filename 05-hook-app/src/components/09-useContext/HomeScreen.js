import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    // Le pido que encuentre la instancia de UserContext en el Ã¡rbol
    const { user } = useContext(UserContext)
    
    return (
        <div>
            <h1> Home Screen </h1>
            <hr />

            <pre className="container">
                { JSON.stringify( user, null, 3) }
            </pre>
        </div>
    )
}
