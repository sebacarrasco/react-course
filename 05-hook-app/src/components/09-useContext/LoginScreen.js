import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext);
    const user = {
        id: 123,
        name: "Seba"
    }

    return (
        <div>
            <h1> Login Screen </h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={ () => setUser(user) }
            >
                Login
            </button>
        </div>
    )
}
