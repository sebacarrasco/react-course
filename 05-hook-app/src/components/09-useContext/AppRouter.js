import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />

                <div className="container">
                    <Switch>

                        {/* El exact es para que sea exactamente ese path */}
                        <Route exact path="/" component={ HomeScreen } />
                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />

                        {/* Nótese que no se le pone path, es decir, si se pone una ruta
                        que no calza con ninguna de las anteriores, entonces se mandará a la homescreen */}
                        {/* <Route component={ HomeScreen } /> */}

                        {/* Lo de arriba es equivalente a hacer: */}
                        <Redirect to="/" />

                    </Switch>
                </div>
            </div>
        </Router>
    )
}
