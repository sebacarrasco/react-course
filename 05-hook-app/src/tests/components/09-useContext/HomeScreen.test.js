import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Tests for <HomeScreen />', () => {

    const user = {
        name: "Sebastián",
        email: "seba@uc.cl"
    }
   
    // El mount es para renderizar todo -> ejemplo: el snapshot
    // con shallow mostraría solo <HomeScreen />, no lo que contiene
    // dentro HomeScreen y no se podría acceder a esto con find
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('should match the snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    });
    

});
