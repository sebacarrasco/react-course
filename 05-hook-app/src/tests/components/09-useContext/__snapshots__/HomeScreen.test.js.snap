// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <HomeScreen /> should match the snapshot 1`] = `
<div>
  <h1>
     Home Screen 
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Sebasti√°n",
   "email": "seba@uc.cl"
}
  </pre>
</div>
`;
